### PR TITLE
feat(webapp): add barcode scanner

### PR DESCRIPTION
**Pull request by Aragur**

### CODE REVIEW

File: src/components/atoms/AtomBarcodeScanner.vue [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+  <div
+    ref="scanner"
+    class="barcode-scanner"
+  />
+</template>
+
+<script lang="ts" setup>
+import Quagga from '@ericblade/quagga2';
+import validbarcode from 'barcode-validator';
+import { ref, watchEffect } from 'vue';
```
Comment (Norowa) [inline src/components/atoms/AtomBarcodeScanner.vue:11]: change import order
```suggestion
import { ref, watchEffect } from 'vue';
import Quagga from '@ericblade/quagga2';
import validbarcode from 'barcode-validator';
```

File: src/components/atoms/AtomBarcodeScanner.vue [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    }
+    ready.value = true;
+    Quagga.onDetected((data) => {
+      if(validbarcode(data.codeResult.code)) {
+        Quagga.stop();
+        emit('detected', data.codeResult.code);
+      }
+    });
+  });
+
```
Comment (Norowa) [inline src/components/atoms/AtomBarcodeScanner.vue:45]: remove blank line
```suggestion
```

**Review by Norowa — CHANGES_REQUESTED** [review]
added 2 suggestion, check them please.

**Review by Malingmax — APPROVED** [review]
looks good

**Review by Norowa — APPROVED** [review]

**Pull request merged**